components/builder-api/src/depot/server.rs

```diff
@@ -2654,6 +2827,94 @@ fn do_cache_response(response: &mut Response) {
     )));
 }
 
+fn set_memcache_package(
+    client: &mut memcache::Client,
+    origin: String,
+    name: String,
+    version: Option<String>,
+    channel: Option<String>,
+    key: &str,
+) -> Result<()> {
+    client.set(&format!("{}/{}", origin, name), key, 0)?;
+    match channel {
+        Some(channel) => {
+            client.set(&format!("{}/{}/{}", channel, origin, name), key.clone(), 0)?;
+            if let Some(set_version) = version {
+                client.set(
+                    &format!("{}/{}/{}/{}", channel, origin, name, set_version),
+                    key.clone(),
+                    0,
+                )?;
+            }
+        }
+        None => {
+            client.set(&format!("unstable/{}/{}", origin, name), key.clone(), 0)?;
+            if let Some(set_version) = version {
+                client.set(
+                    &format!("unstable/{}/{}/{}", origin, name, set_version),
+                    key.clone(),
+                    0,
+                )?;
+            }
+        }
+    };
+    Ok(())
+}
+
+fn get_memcache_package(
+    client: &mut memcache::Client,
+    origin: &str,
+    name: &str,
+    version: Option<&str>,
+    channel: Option<&str>,
```

Comment:
I'm not really liking the optional channel here... the vast majority of the calls we're concerned with have a channel attached... maybe better to always require a channel.. see what happens and what the shape of the code looks like.